### DISCUSSION TITLE
Face matching

### DISCUSSION THREAD

**Discussion by Naofel-eal** [ultralytics/ultralytics#6819] [Q&A]
Hi, I am building an app for person classification in a given video and I would like to know if there is a way to train a Yolov8 model to know if 2 given faces are the same person and to calculate the distance between them. I am actually using Deepface but it's too slow.

**Comment by pderrenger** [comment]
@Naofel-eal hello! For your application, YOLOv8 can be used to detect faces in a video, but it does not inherently perform face matching or calculate the distance between faces. YOLOv8 specializes in object detection, segmentation, and classification tasks.

To achieve face matching, you would typically need a two-step process:

1. **Face Detection**: Use YOLOv8 to detect faces in each frame of the video. You can train a custom YOLOv8 model on a dataset of faces to perform this task efficiently.

2. **Face Matching**: Once faces are detected, you would need a separate facial recognition model to compare the features of the detected faces and determine if they match. This is not a built-in feature of YOLOv8, so you would need to integrate a facial recognition system that can compare the embeddings of the detected faces.

For calculating the distance between detected faces, you could use the bounding box coordinates provided by YOLOv8 after face detection. You can calculate the Euclidean distance between the centers of the bounding boxes to get an approximate distance between faces in pixel space.

If you're looking for improved performance over Deepface, you might consider using a lightweight facial recognition model that can run faster. However, keep in mind that the accuracy of face matching can vary based on the chosen model and the quality of the face embeddings it produces.

For training and deploying YOLOv8 for face detection, you can refer to the [Train](https://docs.ultralytics.com/modes/train) and [Predict](https://docs.ultralytics.com/modes/predict) modes in our documentation. For face matching, you would need to explore facial recognition libraries or frameworks that can be used in conjunction with YOLOv8's detection capabilities.
